retriever/lib/request.js

```diff
@@ -7,28 +9,27 @@
  * @returns {{
  *   clientWalletAddress?: string
  *   rootCid?: string
- *   error?: string
  * }}
  */
 export function parseRequest(request, { DNS_ROOT }) {
   const url = new URL(request.url)
-  console.log({ msg: 'retrieval request', DNS_ROOT, url })
+  console.log('retrieval request', { DNS_ROOT, url })
 
-  if (!url.hostname.endsWith(DNS_ROOT)) {
-    return {
-      error: `Invalid hostname: ${url.hostname}. It must end with ${DNS_ROOT}.`,
-    }
-  }
-  const clientWalletAddress = url.hostname.slice(0, -DNS_ROOT.length)
+  httpAssert(
+    url.hostname.endsWith(DNS_ROOT),
+    400,
+    `Invalid hostname: ${url.hostname}. It must end with ${DNS_ROOT}.`,
+  )
 
+  const clientWalletAddress = url.hostname.slice(0, -DNS_ROOT.length)
   const [rootCid] = url.pathname.split('/').filter(Boolean)
-  if (!rootCid) {
-    return { error: 'Missing required path element: `/{CID}`' }
-  }
 
-  if (!rootCid.startsWith('baga')) {
-    return { error: `Invalid CID: ${rootCid}. It is not a valid CommP root.` }
-  }
+  httpAssert(rootCid, 400, 'Missing required path element: `/{CID}`')
+  httpAssert(
+    rootCid.startsWith('baga'),
+    400,
```

Comment:
Same here. _(Feel free to ignore.)_

```suggestion
    404,
```